File: src/main/scala/chapter01/Student.scala
package chapter01

/**
 * @Author mloine
 * @Description
 * @Date 4:15 PM 2022/1/8
 */
class Student(var name: String, var age: Int) {

  def printInfo(): Unit = {
    println(name + " " + age + " " + Student.school)
  }

}

//引入伴生对象
object Student {

  val school: String = "哈士奇幼儿园"


  def main(args: Array[String]): Unit = {
    val alise = new Student("alise", 20)
    val bob = new Student("bob", 23)

    alise.printInfo()
    bob.printInfo()
  }

}

File: src/main/java/Student.java


/**
 * @Author mloine
 * @Description
 * @Date 4:06 PM 2022/1/8
 */
public class Student {

    private String name;

    private Integer age;

    private static String school = "哈士奇小学";

    public Student(String name, Integer age) {
        this.name = name;
        this.age = age;
    }

    public void printInfo() {
        System.out.println(this.name + " " + this.age + " " + Student.school);
    }

    public static void main(String[] args) {
        Student xueyongkang = new Student("xueyongkang", 20);
        Student bob = new Student("bob", 23);

        xueyongkang.printInfo();
        bob.printInfo();
    }
}
